### PR TITLE
Correct path for ROCMConfig redirect

### PR DESCRIPTION
**Pull request by lawruble13**
The previous redirect in ROCMConfig.cmake did not correctly specify the path to ROCmCMakeBuildTools. This caused failures if the user only specified the path to ROCM, (via something like `ROCM_DIR`), as there were then no correct paths to search for ROCmCMakeBuildTools. I want to create a test for this, but I'm not quite sure how yet, it's in progress.

### CODE REVIEW

File: share/rocm/cmake/ROCMConfig.cmake [hunk 1]
```diff
@@ -2,4 +2,11 @@
...
 
-find_package(ROCmCMakeBuildTools HINTS "${CMAKE_CURRENT_LIST_DIR}")
+get_filename_component(_new_rocmcmakebuildtools_path "${CMAKE_CURRENT_LIST_DIR}" DIRECTORY)
+get_filename_component(_new_rocmcmakebuildtools_path "${_new_rocmcmakebuildtools_path}" DIRECTORY)
+# two directories up is sufficient for windows search, but linux search requires the share directory
+get_filename_component(_new_rocmcmakebuildtools_path_linux "${_new_rocmcmakebuildtools_path}" DIRECTORY)
+
+find_package(ROCmCMakeBuildTools REQUIRED HINTS
+    "${_new_rocmcmakebuildtools_path}"
+    "${_new_rocmcmakebuildtools_path_linux}")
```
Comment (cgmb) [inline share/rocm/cmake/ROCMConfig.cmake]: Should this be `find_package(ROCmCMakeBuildTools HINTS`?

Comment (lawruble13) [reply share/rocm/cmake/ROCMConfig.cmake]: @cgmb I've updated it to pass through the `REQUIRED` and `QUIET` arguments if provided, and the version argument if provided as well. Can you take another look?

Comment (lawruble13) [reply share/rocm/cmake/ROCMConfig.cmake]: As an aside, the `REQUIRED` flag was used originally as `ROCmCMakeBuildTools` should always be installed alongside this file, and so the `REQUIRED` was effectively a short-cut for erroring out if it couldn't be found (indicating that something is wrong).

Comment (lawruble13) [reply share/rocm/cmake/ROCMConfig.cmake]: I looked at your comment prior to your edit and saw that you probably intended to suggest `find_dependency`. I have updated to use this, as it obviates the previous work I did to forward the arguments.

**Comment by lawruble13** [comment]
Note that the failures on macOS are likely unrelated, as the develop branch is also failing on macOS, which should be investigated.

File: share/rocm/cmake/ROCMConfig.cmake (hunk 1 above)
Comment (cgmb) [reply share/rocm/cmake/ROCMConfig.cmake]: Good for catching that. Sorry for being unclear. I tried to add the HINTS with an edit and it seems I accidentally overwrote the core point via a copy/paste error.

**Review by cgmb — APPROVED** [review]
You should perhaps unset your temporary variables at the end of the config script, but this looks good to me.

**Pull request merged**